core/arch/arm/kernel/thread_spmc.c

```diff
@@ -438,6 +438,32 @@ void spmc_handle_partition_info_get(struct thread_smc_args *args,
 		      FFA_PARAM_MBZ, FFA_PARAM_MBZ);
 	cpu_spin_unlock(&rxtx->spinlock);
 }
+
+static void spmc_handle_run(struct thread_smc_args *args)
+{
+	uint16_t endpoint = (args->a1 >> 16) & 0xffff;
+	uint16_t thread_id = (args->a1 & 0xffff);
```

Comment:
can remove parentheses.

I think a `SP_THREAD_ID_MASK` macro would be nice as used at several places.

Reply:
Actually, this part is defined in the FF-A spec. I've added macros for packing and unpacking these fields.